Documentation/getting-started/design.md

```diff
@@ -12,6 +12,31 @@ description: This document describes the design and interaction between the cust
 
 This document describes the design and interaction between the [custom resource definitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/) that the Prometheus Operator manages.
 
+## Internal Architecture Overview
+
+The Prometheus Operator is a Kubernetes Operator that automates the deployment, configuration, and management of Prometheus, Alertmanager, ThanosRuler, and related monitoring resources using Kubernetes Custom Resource Definitions (CRDs). Its core responsibilities include:
+
+- **Custom Resource Management:** The Operator introduces and manages several CRDs, such as `Prometheus`, `Alertmanager`, `ThanosRuler`, `PrometheusAgent`, `ServiceMonitor`, `PodMonitor`, `Probe`, `ScrapeConfig`, `AlertmanagerConfig`, and `PrometheusRule`. These CRDs allow users to declaratively define monitoring infrastructure and scrape configurations in a Kubernetes-native way.
+
+- **Reconciliation Loop:** At the heart of the Operator is a reconciliation loop. The Operator continuously watches for changes to its managed CRDs and associated Kubernetes resources (e.g., StatefulSets, Services, ConfigMaps, Secrets). When a change is detected (creation, update, or deletion), the Operator reconciles the actual state of the cluster to match the desired state specified in the CRDs. This ensures that Prometheus and related components are always correctly configured and running as intended.
```

Comment:
I'd rather link to the Kubernetes documentation which already explains the operator pattern: https://kubernetes.io/docs/concepts/extend-kubernetes/operator/